src/Microsoft.Data/DataFrame.cs

```diff
@@ -133,13 +133,124 @@ public DataFrame Sort(string columnName, bool ascending = true)
             for (int i = 0; i < ColumnCount; i++)
             {
                 BaseColumn oldColumn = Column(i);
-                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(sortIndices, !ascending);
+                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(oldColumn.NullCount, sortIndices, !ascending);
                 Debug.Assert(newColumn.NullCount == oldColumn.NullCount);
                 newColumns.Add(newColumn);
             }
             return new DataFrame(newColumns);
         }
 
+        public enum JoinAlgorithm
+        {
+            Left,
+            Right,
+            Outer,
+            Inner
+        }
+
+        public DataFrame Join(DataFrame other, string leftSuffix = "_left", string rightSuffix = "_right", JoinAlgorithm joinAlgorithm = JoinAlgorithm.Left)
+        {
+            DataFrame ret = new DataFrame();
+            if (joinAlgorithm == JoinAlgorithm.Left)
+            {
+                for (int i = 0; i < ColumnCount; i++)
+                {
+                    BaseColumn newColumn = Column(i).Clone();
+                    newColumn.Name += leftSuffix;
```

Comment:
Shouldn't the suffix only be applied if the names conflict between the two data frames?

Reply:
Oh you're right. I'll change that

Reply:
Done